addons/mrp/mrp.py

```diff
@@ -1120,7 +1119,7 @@ def test_ready(self, cr, uid, ids):
     def _make_production_produce_line(self, cr, uid, production, context=None):
         stock_move = self.pool.get('stock.move')
         proc_obj = self.pool.get('procurement.order')
-        source_location_id = production.product_id.property_stock_production.id
+        source_location_id = (production.routing_id and production.routing_id.location_id and production.routing_id.location_id.id) or p
```

Comment:
The location on the routing is normally an internal location in order to have a production in.  It is not meant to replace Virtual Locations/Production in any way.